pkg/oauth/external/github/github_test.go

```diff
@@ -0,0 +1,201 @@
+package github
+
+import (
+	"bytes"
+	"encoding/json"
+	"errors"
+	"fmt"
+	"io"
+	"net/http"
+	"testing"
+
+	"github.com/RangelReale/osincli"
+	"github.com/openshift/oauth-server/pkg/api"
+	"github.com/openshift/oauth-server/pkg/oauth/external"
+)
+
+func TestGetUserIdentity(t *testing.T) {
+	newGithubIdentityProvider := func(userID uint64, orgs []string) http.RoundTripper {
+		return roundTripperFunc(func(req *http.Request) (*http.Response, error) {
+			body := new(bytes.Buffer)
+
+			switch req.URL.Path {
+			case "/user":
+				if err := json.NewEncoder(body).Encode(struct {
+					ID                 uint64
+					Login, Email, Name string
+				}{
+					ID:    userID,
+					Login: "loginname",
+					Email: "user@example.com",
+					Name:  "drago_tommasone",
+				}); err != nil {
+					panic(err)
+				}
+				return &http.Response{
+					StatusCode: http.StatusOK,
+					Status:     http.StatusText(http.StatusOK),
+					Body:       io.NopCloser(body),
+				}, nil
+
+			case "/user/orgs":
+				type ghOrg struct {
+					ID    uint64
+					Login string
+				}
+				ghOrgs := make([]ghOrg, len(orgs))
+				for i := range orgs {
+					ghOrgs[i] = ghOrg{
+						ID:    999,
+						Login: orgs[i],
+					}
+
+				}
+
+				if err := json.NewEncoder(body).Encode(ghOrgs); err != nil {
+					panic(err)
+				}
+				return &http.Response{
+					StatusCode: http.StatusOK,
+					Status:     http.StatusText(http.StatusOK),
+					Body:       io.NopCloser(body),
+				}, nil
+
+			default:
+				return nil, fmt.Errorf("this fixture does not serve the requested path: %s", req.URL.Path)
+			}
+		})
+	}
+
+	type checkFunc func(api.UserIdentityInfo, error) error
+	checks := func(fns ...checkFunc) []checkFunc { return fns }
```

Comment:
why don't you just add the functions directly to a slice?

Reply:
it looks and reads cleaner :smile: 

I enjoy such constructs quite a lot :sweat_smile:

Reply:
done